### DISCUSSION TITLE
comments and colours in history

### DISCUSSION THREAD

**Discussion by underlow** [marlonrichert/zsh-autocomplete#475] [Q&A] [answered]
Couple more questions. 
1. Is it possible to color history same as command line?
2. is it possible to add comment to history? 
If I have command with comment in history like 
`killall bird # restart iCloud process` it shows in history without comment. It searches through comments though
<img width="378" alt="image" src="https://user-images.githubusercontent.com/15636461/187034270-7bacfb94-2f0a-4fa1-bf10-a7366e7764df.png">

**Comment by marlonrichert** [comment, marked as answer]
> 1. Is it possible to color history same as command line?

No, not trivially. However, while you step through the results, your command line shows the current selection with full syntax highlighting.

> 2. Is it possible to add comment to history?

Yes, that's easily possible. It's an oversight on my part that causes them to be filtered out in the displayed results, but that's easily fixed. I've opened #476 for it.

**Comment by kinland** [comment]
It would be very cool to be able to apply `zsh-users/zsh-syntax-highlighting` or `zdharma-continuum/fast-syntax-highlighting` to the history results, not just to the current selection

**Reply by marlonrichert** [reply]
Feel free to write your own plugin for that. 😉
